src/app/routes/share.ts

```diff
@@ -1,36 +1,275 @@
+        res.status(200).json(results);
+      })
+      .catch((err: Error) => {
+        res.status(500).json({
+          error: err.message
+        });
+      });
+  }
+
+  public async getSharedDirectoryFolders(req: Request, res: Response): Promise<void> {
+    const { token, directoryId, offset, limit } = req.query;
+
+    if (Validator.isInvalidString(token)) {
+      throw createHttpError(400, 'Token must be a valid string');
+    }
+
+    if (Validator.isInvalidPositiveNumber(directoryId)) {
+      throw createHttpError(400, 'Directory ID is not valid');
+    }
+
+    if (Validator.isInvalidUnsignedNumber(offset)) {
+      throw createHttpError(400, 'Offset is not valid');
+    }
+
+    if (Validator.isInvalidPositiveNumber(limit)) {
+      throw createHttpError(400, 'Limit is not valid');
+    }
+
+    return this.services.Share.getSharedDirectoryFolders(directoryId, Number(offset), Number(limit), token)
+      .then((results: unknown) => {
+        res.status(200).json(results);
+      })
+      .catch((err: Error) => {
+        res.status(500).json({
+          error: err.message
+        });
+      });
+  }
+
+  private logReferralError(userId: unknown, err: Error) {
+    if (!err.message) {
+      return this.logger.error('[STORAGE]: ERROR message undefined applying referral for user %s', userId);
+    }
+
+    if (err instanceof ReferralsNotAvailableError) {
+      return;
+    }
+
+    return this.logger.error('[STORAGE]: ERROR applying referral for user %s: %s', userId, err.message);
+  };
 }
 
-export default (router: Router, service: any) => {
-  const controller = new ShareController(service);
+export default (router: Router, service: any,) => {
+  const Logger = logger.getInstance();
+  const controller = new ShareController(service, Logger);
   const sharedAdapter = sharedMiddlewareBuilder.build(service);
 
-  router.get('/share/list', passportAuth, sharedAdapter, controller.listShares.bind(controller));
-  router.get('/share/:shareId/folder/:folderId', controller.getSharedFolderSize.bind(controller));
+  router.get('/share/list', passportAuth, sharedAdapter,
+    controller.listShares.bind(controller)
+  );
+  router.get('/share/:shareId/folder/:folderId',
+    controller.getSharedFolderSize.bind(controller)
+  );
+  router.post('/storage/share/file/:id', passportAuth, sharedAdapter,
+    controller.generateShareFileToken.bind(controller)
+  );
+  router.post('/storage/share/folder/:id', passportAuth, sharedAdapter,
+    controller.generateShareFolderToken.bind(controller)
+  );
+  router.get('/storage/share/:token',
+    controller.getShareFileInfo.bind(controller)
+  );
+  router.get('/storage/shared-folder/:token',
+    controller.getShareFolderInfo.bind(controller)
+  );
+  router.get('/storage/share/down/files',
+    controller.getSharedDirectoryFiles.bind(controller)
+  );
+  router.get('/storage/share/down/folders',
+    controller.getSharedDirectoryFolders.bind(controller)
+  );
 };
```

Comment:
`generateShareFileToken` (`createFileShare` could be a better naming) -> `POST /share/file/:id`
`generateShareFolderToken` (`createFolderShare` could be a better naming) -> `POST /share/folder/:id`
`getShareFileInfo` (`getFileShareInfo`/`getSharedFileInfo` could be a better naming) -> `GET /share/file/:id`
`getShareFolderInfo` (`getFolderShareInfo`/`getSharedFolderInfo` could be a better naming -> `GET /share/folder/:id`
`getSharedDirectoryFiles` -> `GET /share/:shareId/folder/:folderId/files` *
`getSharedDirectoryFolders` -> `GET /share/:shareId/folder/:folderId/folders` *

As you have the entire share due to calling `getSharedXInfo`, that id could be used as `shareId` here